tools/roxvet/analyzers/protoptrs/analyzer.go

```diff
@@ -0,0 +1,78 @@
+package protoptrs
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+	"regexp"
+
+	"github.com/stackrox/rox/tools/roxvet/common"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+var (
+	protoMsgType = types.NewInterfaceType(
+		[]*types.Func{
+			types.NewFunc(token.NoPos, nil, "ProtoMessage", types.NewSignatureType(nil, nil, nil, types.NewTuple(), types.NewTuple(), false)),
+		},
+		nil).Complete()
+
+	protoTypesRegex = regexp.MustCompile(`^github\.com/stackrox/rox/generated[./]|^github\.com/gogo/protobuf/types[./]`)
+)
+
+// Analyzer is the go vet entrypoint
+var Analyzer = &analysis.Analyzer{
+	Name:     "protoptrs",
+	Doc:      "checks that protobuf message types are only used as pointers",
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	inspectResult := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{
+		(*ast.FuncType)(nil),
+		(*ast.StructType)(nil),
+	}
+	common.FilteredPreorder(inspectResult, common.Not(common.IsGeneratedFile), nodeFilter, func(n ast.Node) {
```

Comment:
out of curiosity: do we also get `ast.FuncType/StructType` from the go template file in `generate-helpers/pg-table-bindings/migration.go.tpl`? Since you fixed the postgres migrations and some of them _should_ be detected as generated files.

If that's the case, this is quite neat to get it from the template files!

Reply:
I wish it worked that way, but unfortunately no, vet-tools simply look at actual Go code, not the templates. Those fixes were applied manually by me to the templates, after my IDE warned me that the file I was about to edit (based on the `roxvet` finding) was generated.